sys/core/src/TxsCore.hs

```diff
@@ -1053,62 +1082,63 @@ txsNComp (TxsDefs.ModelDef insyncs outsyncs splsyncs bexp) =  do
 -- ----------------------------------------------------------------------------------------- --
 
 -- | LPE transformation by Carsten Ruetz
-txsLPE :: TxsDefs.BExpr                     -- ^ behaviour expression, to be transformed;
-                                            --   shall be a process instantiation.
-       -> IOC.IOC (Maybe TxsDefs.BExpr)     -- ^ transformed process instantiation
-txsLPE bexpr  =  do
+txsLPE :: Either TxsDefs.BExpr TxsDefs.ModelId   -- ^ either a behaviour expression (that shall
```

Comment:
I couldn't understand the reason to use `Either` here instead of having two different functions. Since by convention a `Left` value holds an error case, this implementation can be confusing at usage points. Splitting this function can also save some Right/Left packing-unpacking.

Reply:
'Either' can be used for many things.
When used for error handling indeed one has the right value and the left error. But I assume here it is used as a union of two types.